src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs

```diff
@@ -294,25 +296,28 @@ private void ReadRle<TPixel>(BmpCompression compression, Buffer2D<TPixel> pixels
             where TPixel : struct, IPixel<TPixel>
         {
             TPixel color = default;
-            using (Buffer2D<byte> buffer = this.memoryAllocator.Allocate2D<byte>(width, height, AllocationOptions.Clean))
-            using (Buffer2D<bool> undefinedPixels = this.memoryAllocator.Allocate2D<bool>(width, height, AllocationOptions.Clean))
+            using (IMemoryOwner<byte> buffer = this.memoryAllocator.Allocate<byte>(width * height, AllocationOptions.Clean))
```

Comment:
If i would use `Buffer2d` i would need to process row wise, which is not really suitable in the current implementation. One run length can span over multiple rows.

I think it is possible to change RLE decompression to work with rows, but it would be a complete overhaul of the implementation. I dont think at the moment it is worth the effort. Mainly because i dont see the usecase of very large RLE bitmaps.